project_timesheet_holidays_contract/models/hr_employee.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    def _get_datetime_date(self, date):
+        return date if not isinstance(date, datetime) else date.date()
+
+    def _get_calendar(self, date_from=None):
+        res = super()._get_calendar(date_from)
+        if not date_from:
+            return res
```

Comment:
You only need to call `super()` if:
- `date_from` is not defined
- no calendar is found according to the logic below

Therefore, calling `super()` at the beginning of the method, _before_ you check the conditions mentioned above, just doesn't make any sense. If you call `super()` now:
- if you don't find any calendar later on, you'll end up calling `super()` again at line 45), so Odoo will perform twice the same method
- if you do find a calendar, the result of the `super()` call will be ignored, making the execution of the `super()` itself useless
- the only case in which you actually use the `super()`'s result is when `date_from` is not set

Instead, what you should have is:
```
        if not date_from:  # no `date_from` => call `super()`
            return super()._get_calendar(date_from)
        ... # try to retrieve calendars
        if calendar:
            return calendar  # calendar found => no need to call `super()`
        return super()._get_calendar(date_from)  # no calendar found => call `super()`
```